github.js

```diff
@@ -1,40 +1,46 @@
 var https = require('https');
+var rp = require('request-promise');
+var Rx = require('rx');
 
-exports.request = function (path, callback) {
+var request = function (path) {
 
-  var authorization = "";
-  if (process.env.GITHUB_TOKEN == null)
+  var headers = {
+      'User-Agent': "Up For Grab Data Service"
+  };
+
+  if (process.env.GITHUB_TOKEN != null)
   {
-    console.log("No environment variable set for GitHub token, unauthenticated client is very restricted...");
-  } else {
-    console.log("Found GITHUB_TOKEN variable, setting header...");
-    authorization = 'Token ' + process.env.GITHUB_TOKEN;
+     headers['Authorization'] = "Token " + process.env.GITHUB_TOKEN;
   }
 
   var options = {
-    host: 'api.github.com',
-    path: path,
-    headers: {
-      'Authorization': authorization,
-      'User-Agent': "Up For Grab Data Service"
-    }
+      uri: 'https://api.github.com' + path,
+      headers: headers,
+      json: true
   };
 
-  var str = '';
+  var promise = rp(options);
```

Comment:
I switched over from the plain `http` package for two reasons:

 - I didn't really need to do anything fancy inside the response callbacks (see below) - I just wanted the response body. And I know it'll always be JSON, so I moved up the abstraction stack.
 - I wanted to use a promise here rather than pass in a callback.